app/services/state_file/send_post_deadline_reminder_service.rb

```diff
@@ -0,0 +1,39 @@
+module StateFile
+  class SendPostDeadlineReminderService
+    BATCH_SIZE = 10
+    HOURS_AGO = 24
+
+    def self.run
+      cutoff_time_ago = HOURS_AGO.hours.ago
+      intakes_to_notify = []
+
+      ApplicationRecord::STATE_INTAKE_CLASS_NAMES.each do |base_class|
+        class_object = base_class.constantize
+        intakes_to_notify += class_object.left_joins(:efile_submissions)
+                                         .where(efile_submissions: { id: nil })
+                                         .where("#{base_class.underscore.pluralize}.created_at < ?", cutoff_time_ago)
+                                         .where.not(email_address: nil)
+                                         .where.not(email_address_verified_at: nil)
+                                         .where(unsubscribed_from_email: false)
+                                         .where("#{base_class.underscore.pluralize}.message_tracker #> '{messages.state_file.post_deadline_reminder}' IS NULL")
+                                         .select do |intake|
+          if intake.message_tracker.present? && intake.message_tracker["messages.state_file.finish_return"]
+            finish_return_date = Date.parse(intake.message_tracker["messages.state_file.finish_return"])
```

Comment:
I would rename: `finish_return_message_sent_time`

Reply:
Ended up going with `finish_return_msg_sent_time` since Rubocop said it was too long.